### ISSUE TITLE
vcluster delete --all Command

### ISSUE THREAD

**Issue by danielhelfand** [loft-sh/vcluster#363]
When using the `vcluster` CLI, it would be nice to support an option to delete all virtual clusters on the host cluster. Something like `vcluster delete --all` would be helpful in really quickly cleaning up clusters without having to remember specific cluster names/specify all clusters.

Something to consider would be the best way to fetch all the namespaces where virtual clusters are deployed to in order to pass on to helm. Also, the `--delete-namespace` option would need a way to have all the relevant namespaces passed to it.

The other things to consider would be failure scenarios. Should the command continue if an error is encountered deleting a particular virtual cluster?

Not sure if there would be additional considerations for nested virtual clusters. 

Looking forward to thoughts on this!

**Comment by FabianKramm** [comment]
@danielhelfand thanks for creating this issue! Not sure if we should implement this command as it might have devastating consequences if ran on accident. Besides that as you already mentioned there are a couple of pitfalls like namespace deletion, which can be even more devastating if a vcluster was installed into kube-system for example and vcluster is trying to delete the complete namespace.

I think it might be better to have `vcluster list` printing the vclusters in a way you could easily pass those to `helm delete` or `kubectl delete namespace`, which makes it still easy to use but circumvents the potential pitfalls for now.

**Comment by FabianKramm** [comment]
@danielhelfand yes having a confirm might make sense, another thought would be to label namespaces with a vcluster label during creation to later identify them that they are safe to delete, which would prevent accidental deletion of important namespaces like kube-system.

**Comment by sarojregmi200** [comment]
@FabianKramm since no one is assigned to this Issue, can I work on this issue? 
I want to give it a try.

It seems to be a old issue but the labels indicate that it is something that we can work on now.
I would love to  see and know if it is something that the  community can work upon.
Thanks.

**Comment by SEEDART007** [comment]
@danielhelfand hey it seems the assignees are not working in this issue can I work?

**Comment by SEEDART007** [comment]
> @SEEDART007 you can take over the issue. I forgot about it, and am not available to work on it.

That's fine I'll start working from tomorrow in this issue

**Comment by usersina** [comment]
In case it helps anyone, I cufrrently do this with `zsh`

```bash
vcluster list | tail -n +3 | while IFS='|' read -r NAME NS _rest; do
  NAME=${NAME// /}
  NS=${NS// /}
  if [[ -n "$NAME" && "$NAME" != ---* ]]; then
    echo "Deleting $NAME in namespace $NS..."
    vcluster delete "$NAME" -n "$NS"
  fi
done
```
